packages/aws-cdk-lib/aws-ecs/lib/fargate/fargate-task-definition.ts

```diff
@@ -168,11 +170,16 @@ export class FargateTaskDefinition extends TaskDefinition implements IFargateTas
     }
 
     if (props.pidMode) {
+      if (!props.runtimePlatform?.operatingSystemFamily) {
+        throw new Error('Specifying \'pidMode\' requires that operating system family also be provided.');
+      }
       if (props.runtimePlatform?.operatingSystemFamily?.isWindows()) {
         throw new Error('\'pidMode\' is not supported for Windows containers.');
       }
```

Comment:
nit: I think what we care here is it should be linux. So we can move the `isLinux()` check up here and update the error message.
```
if (!props.runtimePlatform?.operatingSystemFamily?.isLinux())
```

Reply:
From what I understand, that check was put there to verify that the user isn't trying to set pidMode on a Windows container.  The documentation says it's not supported.  Given that, do you think it would be worth keeping?

![image](https://github.com/aws/aws-cdk/assets/88447527/91051edc-cb6b-44f5-88dc-d2cfdc92da70)

https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#task_definition_pidmode

Reply:
Got it. I'm fine keeping the check here.